builtin/docker/pull/builder.go

```diff
@@ -126,15 +131,25 @@ func (b *Builder) Config() (interface{}, error) {
 }
 
 // Build
-func (b *Builder) Build(
-	ctx context.Context,
-	ui terminal.UI,
-	src *component.Source,
-	log hclog.Logger,
-) (*wpdocker.Image, error) {
+func (b *Builder) Build(args struct {
+	// We use the struct form of arguments so that we can access named
+	// values (such as "HasRegistry").
+	argmapper.Struct
+
+	Ctx         context.Context
+	UI          terminal.UI
+	Log         hclog.Logger
+	HasRegistry bool
+}) (*wpdocker.Image, error) {
```

Comment:
While nifty, I think we should probably pull this out as a named struct. The pattern of arg structs in grpc is well known and it's less surprising than seeing an anon struct in a params listing.